vendor/update-re.sh

```diff
@@ -10,18 +10,26 @@ trap "rm -rf $TMP" EXIT
 rm -rf re
 mkdir -p re/src
 
-(cd $TMP && opam source re.$version)
+(
+    cd $TMP
+    git clone https://github.com/ocaml/ocaml-re.git
+    cd ocaml-re
+    git checkout $version
+)
 
-SRC=$TMP/re.$version
+SRC=$TMP/ocaml-re
 
-cp -v $SRC/LICENSE re
+cp -v $SRC/LICENSE.md re
+cp -v $SRC/lib/*.{ml,mli} re/src/
 
-for m in re re_automata re_cset re_fmt; do
-    for ext in ml mli; do
-        fn=$SRC/lib/$m.$ext
-        [[ -f $fn ]] && cp -v $fn re/src
-    done
-done
+echo "include Re" > re/src/dune_re.ml
+cat >re/src/dune <<EOF
+(subdir re
+ (library
+  (name        dune_re)
+  (public_name dune-private-libs.dune_re)
+  (synopsis "Internal Dune library, do not use!")))
+EOF
 
-git checkout re/src/jbuild
+git checkout re/src/dune
```

Comment:
I expect that this would restore the previous `re/src/dune`, erasing the effect the above `cat` command. The idea of the `git checkout re/src/dune` is to have the `dune` file manually maintained in the repo. We can either git checkout this file, or write it via the script, but we can't do both.

Reply:
Good spot. I got rid of it.

Reply:
@jeremiedimino this PR is ready